pkg/devfile/validate/generic/errors.go

```diff
@@ -0,0 +1,136 @@
+package generic
+
+import "fmt"
+
+// InvalidEventError returns an error if the devfile event type has invalid events
+type InvalidEventError struct {
+	eventType string
+	errorMsg  string
+}
+
+func (e *InvalidEventError) Error() string {
+	return fmt.Sprintf("%s type events are invalid: %s", e.eventType, e.errorMsg)
+}
+
+// DuplicateCommandError returns an error if the command is duplicate
+type DuplicateCommandError struct {
+	commandId string
+}
+
+func (e *DuplicateCommandError) Error() string {
+	return fmt.Sprintf("devfile has duplicate command IDs %q", e.commandId)
+}
+
+// ExecCommandMissingComponentError returns an error if the exec command does not have a component
+type ExecCommandMissingComponentError struct {
+	commandId string
+}
+
+func (e *ExecCommandMissingComponentError) Error() string {
+	return fmt.Sprintf("exec command %q must reference a component", e.commandId)
+}
+
+// ExecCommandMissingCommandLineError returns an error if the exec command does not have a command line
+type ExecCommandMissingCommandLineError struct {
+	commandId string
+}
+
+func (e *ExecCommandMissingCommandLineError) Error() string {
+	return fmt.Sprintf("exec command %q must have a command", e.commandId)
+}
```

Comment:
should this all be covered in a single error like `MissingInformationError`?

Reply:
This error refactoring could be done in separate PR, for now its ok.

Reply:
I've already refactored, just need to push my changes